README.md

```diff
@@ -1,50 +1,154 @@
-[PagerDuty][2] incidents, for alerting operators. With this handler,
-[Sensu][1] can trigger and resolve PagerDuty incidents.
+[![Bonsai Asset Badge](https://img.shields.io/badge/Bonsai-Download%20Me-brightgreen.svg?colorB=89C967&logo=sensu)](https://bonsai.sensu.io/assets/sensu/sensu-pagerduty-handler)
+[![ Build Status](https://travis-ci.org/sensu/sensu-pagerduty-handler.svg?branch=master)](https://travis-ci.org/sensu/sensu-pagerduty-handler)
 
-## Installation
+- [Overview](#overview)
+- [Configuration](#configuration)
+  - [Environment Variables](#environment-variables)
+- [Setup](#setup)
+- [Examples](#examples)
+- [Acknowledgements](#acknowledgements)
 
-Download the latest version of the sensu-pagerduty-handler from [releases][4],
-or create an executable script from this source.
+## Overview
 
-From the local path of the sensu-pagerduty-handler repository:
-```
-go build -o /usr/local/bin/sensu-pagerduty-handler main.go
-```
+The Sensu Go PagerDuty Handler is a [Sensu Go event handler][3] which manages
+[PagerDuty][2] incidents for alerting operators. With this handler,
+[Sensu][1] can trigger and resolve PagerDuty incidents.
 
 ## Configuration
 
-Example Sensu Go handler definition:
+You can use either command line flags or environment variables to configure the Sensu Pagerduty Handler. The environment variables can be in either the handler definition or in the Sensu backend service environment.
+
+**Note:** Make sure to set the `PAGERDUTY_TOKEN` environment variable for sensitive credentials in production to prevent leaking into system process table. Please remember command arguments can be viewed by unprivileged users using commands such as `ps` or `top`. The `--token` argument is provided as an override primarily for testing purposes.
+
+Pagerduty Token        | |
+---------------------|-------------------------------
+description          | The PagerDuty V2 API authentication token.
+long                 | token
+short                | t
+env variable         | PAGERDUTY_TOKEN
+required             | true
+type                 | String
+default              | null
+example              | `PAGERDUTY_TOKEN=f88b6c03e43d3aa939959a8f47fjge`
+
+Pagerduty Dedup Key        | |
+---------------------|-------------------------------
+description          | The Sensu event label specifying the PagerDuty V2 API deduplication key.
+long                 | dedup-key
+short                | d
+env variable         | PAGERDUTY_DEDUP_KEY
+required             | false
+type                 | String
+default              | null
+example              | `PAGERDUTY_DEDUP_KEY=test_label`
+
+Pagerduty Dedup Key Template        | |
+---------------------|-------------------------------
+description          | The PagerDuty V2 API deduplication key template.
+long                 | dedup-key-template
+short                | k
+env variable         | PAGERDUTY_DEDUP_KEY_TEMPLATE
+required             | false
+type                 | String
```

Comment:
Is this a String or some Sensu native type since it is looking of the entity and check names? I might need more context on what a dedup key for PD is.